app/api/v1/endpoints/control/attacks.py

```
@@ -0,0 +1,769 @@
+    Cannot update attacks that are currently running.
+    The user must have access to the campaign's project.
+    """
+    try:
+        # Validate access and get attack
+        attack = await _get_attack_with_access_check(attack_id, current_user, db)
+
+        # Prevent updating running attacks
+        if attack.state == AttackState.RUNNING:
+            raise InvalidResourceStateError(
+                detail="Cannot update attack while it is running. Stop the attack first."
+            )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)